File: 658_B.cpp
/*
 * https://codeforces.com/contest/1382/problem/B
 * Sequential Nim.
 */

# include<iostream>
# include<vector>

void solve(std::vector<int>& a)
{
    bool first = true;

    for(int i = 0; i < a.size()-1; i++)
    {
        if(a[i] == 1) first = not first;
        else break;
    }

    if(first) std::cout<<"First"<<std::endl;
    else std::cout<<"Second"<<std::endl;

    return;
}

int main()
{
    int t, n;
    std::cin>>t;

    while(t--)
    {
        std::cin>>n;

        std::vector<int> a(n);
        int val, index = 0;

        while(n--)
        {
            std::cin>>val;
            a[index++] = val;
        }
        solve(a);
    }

    return 0;
}

File: 658_B.py
"""
 * https://codeforces.com/contest/1382/problem/B
 * Sequential Nim.
"""

def solve(a, n):
    first = True

    for i in range(n-1):
        if a[i] == 1: first = not first
        else: break

    if first: print("First")
    else: print("Second")

if __name__ == "__main__":
    t = int(input())

    while(t):
        n = int(input())
        a = [(lambda x: int(x))(x) for x in input().split()]
        solve(a, n)
        t -= 1
